--- pkg/models/server.go
+++ pkg/models/server.go
@@ -9,16 +9,11 @@
	IP   string     `json:"ip" gorm:"column:ip"`
	Port int        `json:"port" gorm:"column:port"`
	Type ServerType `json:"type" gorm:"column:type;index"`
+	Username string `json:"username,omitempty" gorm:"column:username"`
+	Password string `json:"password,omitempty" gorm:"column:password"`

-	// Windows Specific
-	WindowsUsername string `json:"username,omitempty" gorm:"column:windows_username"`
-	WindowsPassword string `json:"password,omitempty" gorm:"column:windows_password"`
-
	// VMware Specific
	VMwareThumbPrint string   `json:"vmware_thumb_print,omitempty" gorm:"column:vmware_thumb_print"`
-	VMwareUsername   string   `json:"username,omitempty" gorm:"column:vmware_username"`
-	VMwarePassword   string   `json:"password,omitempty" gorm:"column:vmware_password"`
-	// For arrays of arrays in pg, one approach is JSONB, which is highly compatible.
	VMwareVMs        [][]string `json:"vmware_vms,omitempty" gorm:"column:vmware_vms;type:jsonb;serializer:json"`
 }
